india_compliance/patches/post_install/remove_tax_category.py

```diff
@@ -0,0 +1,154 @@
+        {
+            "fieldname": "is_reverse_charge",
+            "label": "Is Reverse Charge",
+            "fieldtype": "Check",
+            "insert_after": "is_inter_state",
+            "print_hide": 1,
+        },
+    ],
+    "Supplier": [
+        {
+            "fieldname": "is_reverse_charge",
+            "label": "Is Reverse Charge by Default",
+            "fieldtype": "Check",
+            "insert_after": "gst_transporter_id",
+            "depends_on": "eval:doc.gst_category == 'Registered Regular'",
+            "translatable": 0,
+        }
+    ],
+}
+CUSTOM_FIELDS_TO_DELETE = {
+    DOCTYPE: [
+        {
+            "fieldname": "is_inter_state",
+        },
+        {
+            "fieldname": "is_reverse_charge",
+        },
+        {
+            "fieldname": "tax_category_column_break",
+        },
+        {
+            "fieldname": "gst_state",
+        },
+    ],
+}
+DEFAULT_CATEGORIES = [
+    "In-State",
+    "Out-State",
+    "Reverse Charge In-State",
+    "Reverse Charge Out-State",
+    "Registered Composition",
+]
+
+
+def execute():
+    if not frappe.flags.in_install:
+        create_custom_fields(CUSTOM_FIELDS_TO_CREATE)
+        set_default_gst_settings()
+
+    update_tax_templates()
+    if not opt_out_needed():
+        return switch_to_simplified_tax()
+
+    click.secho(
+        "Looks like you are using tax categories. You can opt into simplified tax for India from GST Settings.",
+        color="yellow",
+    )
+
+
+def set_default_gst_settings():
+    settings = frappe.get_doc("GST Settings")
+    settings.db_set(
+        {
+            "reverse_charge_for_unregistered_purchase": 1,
+            "reverse_charge_threshold": 5000,
+        }
+    )
+
+
+def update_tax_templates():
+    show_alert = False
+    for doctype in TAX_TEMPLATES:
+        doc_list = frappe.get_all(doctype, pluck="name")
+
+        for docname in doc_list:
+            doc = frappe.get_doc(doctype, docname)
+            if not doc.tax_category:
+                continue
+
+            category = frappe.db.get_value(
+                DOCTYPE,
+                {"name": doc.tax_category},
+                ["is_inter_state", "is_reverse_charge", "gst_state"],
+                as_dict=True,
+            )
+
+            if category.gst_state:
+                show_alert = True
+
+            update_values = {"is_inter_state": category.is_inter_state}
+            if doctype == "Purchase Taxes and Charges Template":
+                update_values["is_reverse_charge"] = category.is_reverse_charge
+
+            doc.db_set(update_values)
+
+    if show_alert:
+        click.secho(
+            "Looks like you are using State wise Tax Categories. They are no longer required and hence not supported.",
+            color="yellow",
+        )
+
+
+def opt_out_needed():
+    categories = frappe.get_all(DOCTYPE, pluck="name")
+    for category in categories:
+        if category not in DEFAULT_CATEGORIES:
```

Comment:
In-state and Out-state are just titles and there might be cases that users might be using the old config with tax category with some other non-standard title, there are chances such users might be affected by this patch

Reply:
This patch doesn't delete GST Category (as set in various docs) if configured in Customer, Supplier, Address, or Item Tax.
`opt_out_needed` makes a check for the same.

Users in such scenarios can manually trigger the delete (Switch to Simplified Tax Setup - Screenshot 1) from GST Settings.